File: app/src/main/java/com/lumiyaviewer/lumiya/slproto/objects/SLPrimObjectDisplayInfoWithChildren.java
package com.lumiyaviewer.lumiya.slproto.objects;

import com.google.common.collect.ImmutableList;
import com.lumiyaviewer.lumiya.slproto.objects.SLObjectDisplayInfo;
import javax.annotation.Nonnull;

public class SLPrimObjectDisplayInfoWithChildren extends SLPrimObjectDisplayInfo implements SLObjectDisplayInfo.HasChildrenObjects {
    @Nonnull
    public final ImmutableList<SLObjectDisplayInfo> children;
    private final boolean implicitlyAdded;

    public SLPrimObjectDisplayInfoWithChildren(SLObjectInfo sLObjectInfo, float f, @Nonnull ImmutableList<SLObjectDisplayInfo> immutableList, boolean z) {
        super(sLObjectInfo, f);
        this.children = immutableList;
        this.implicitlyAdded = z;
    }

    @Nonnull
    public ImmutableList<SLObjectDisplayInfo> getChildren() {
        return this.children;
    }

    public boolean isImplicitlyAdded() {
        return this.implicitlyAdded;
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/slproto/objects/SLPrimObjectDisplayInfoWithChildren.kt
package com.lumiyaviewer.lumiya.slproto.objects
import java.util.*

import com.google.common.collect.ImmutableList
import com.lumiyaviewer.lumiya.slproto.objects.SLObjectDisplayInfo
import javax.annotation.Nonnull

class SLPrimObjectDisplayInfoWithChildren : SLPrimObjectDisplayInfo : SLObjectDisplayInfo.HasChildrenObjects {
    @Nonnull
    ImmutableList<SLObjectDisplayInfo> children
    private Boolean implicitlyAdded

    SLPrimObjectDisplayInfoWithChildren(SLObjectInfo sLObjectInfo, Float f, @Nonnull ImmutableList<SLObjectDisplayInfo> immutableList, Boolean z) {
        super(sLObjectInfo, f)
        this.children = immutableList
        this.implicitlyAdded = z
    }

    @Nonnull
    ImmutableList<SLObjectDisplayInfo> getChildren() {
        return this.children
    }

    Boolean isImplicitlyAdded() {
        return this.implicitlyAdded
    }
}
